### ISSUE TITLE
[docs] Add more details on implementing SEO techniques in Gatsby

### ISSUE THREAD

**Issue by marcysutton** [gatsbyjs/gatsby#14125]
## Summary

In PR #10780, @DSchau proposed a really great piece of content on the blog on implementing SEO techniques in Gatsby. We decided to close that PR in favor of a new issue for a Reference Guide instead of a blog post–but the content from the PR could be copied and slightly modified. I think it makes the most sense to add it as an Advanced Tutorial (dependent on #14071) and link to that from the [SEO docs page](https://www.gatsbyjs.org/docs/seo/).

To fit in with tutorials and docs, any mentions of "us" or "we" should change to "you" to directly address the reader (more in the [Gatsby Style Guide](https://www.gatsbyjs.org/contributing/gatsby-style-guide/#use-you-as-the-pronoun))

This will be a great how-to for adding SEO stuff to Gatsby sites, especially explaining what's in some of the starters. As a tutorial instead of a blog post, it will be easier to find after more content is added since it will be more discoverable through the docs.

**Comment by KyleAMathews** [comment]
Here's a good reference for adding JSON-LD structured data https://nystudio107.com/blog/annotated-json-ld-structured-data-examples

**Comment by gmossong** [comment]
Would be good to see techniques for not only the usual stuff like title, description, Facebook and Twitter meta tags but also [structured data](https://developers.google.com/search/docs/guides/intro-structured-data).

**Comment by FlorianDr** [comment]
I would like to work on this. I started moving the content from the blog post to the tutorials folder.

**Comment by marcysutton** [comment]
That sounds great, @FlorianDr!! Let us know if you have any questions.

**Comment by gmossong** [comment]
I have [a question](https://stackoverflow.com/q/58808915/11722922) where I'm hoping I can get some input on how to best handle [rich snippets and structured data](https://developers.google.com/search/docs/guides/search-gallery). I think this is a really important part of SEO and would be great to include in the docs.

**Comment by marcysutton** [comment]
Hi @FlorianDr, I wanted to check up on this issue. Are you still interested in contributing a Gatsby tutorial on implementing SEO techniques? Also, any thoughts on the question about rich snippets and Structured Data?

**Comment by FlorianDr** [comment]
Hi @marcysutton I'm so sorry for not responding or working on this issue for such a long time. I kept postponing it. That's bad behavior. 

I will work on it this week and open a PR.
